packages/material-ui/src/ListItem/ListItem.js

```diff
@@ -107,11 +109,23 @@ const ListItem = React.forwardRef(function ListItem(props, ref) {
     dense: dense || context.dense || false,
     alignItems,
   };
+  const listItemRef = React.useRef();
+  React.useLayoutEffect(() => {
+    if (autoFocus) {
+      listItemRef.current.focus();
+    }
+  }, [autoFocus]);
 
   const children = React.Children.toArray(childrenProp);
   const hasSecondaryAction =
     children.length && isMuiElement(children[children.length - 1], ['ListItemSecondaryAction']);
 
+  const handleOwnRef = React.useCallback(refArg => {
```

Comment:
Nit: 
```suggestion
  const handleOwnRef = React.useCallback(instance => {
```
instead of the generic `refArg`. 

I know we use a similar naming (`refArg`) in other places but that isn't really descriptive of that the value actually holds.